File: labs/J10_Observer/python/startingpoint/filedownloader.py
from logger import Logger

class FileDownloader():
    logger=None
    progressreporter = None
    def __init__(self,logger,progressreporter):
        self.logger = logger
        self.progressreporter=progressreporter

    def download(self,file):
        beginMessage = 'beginning load...'
        self.progressreporter.update(beginMessage)
        self.logger.info(beginMessage)

        for i in range(0, 10):
            message = 'percent complete' + str(i * 10)
            self.progressreporter.update(message)
            self.logger.info(message)

        endmessage = 'done with download'
        self.progressreporter.update(endmessage)
        self.logger.info(endmessage)

File: labs/J10_Observer/java/complete/app/src/main/java/ObserverComplete/FileDownloader.java
package ObserverComplete;

public class FileDownloader {
    private final ProgressReporter reporter = new ProgressReporter();
    private final Logger logger = new Logger(); 


    public void download(String file){
        final String beginMessage = "Beginning to download.";
        this.logger.info(beginMessage);
        this.reporter.update(beginMessage);

        for(int i =0; i<10;i++)
		{
			final String message = "Percent complete: " + String.valueOf(i*10);
			logger.info(message);
			reporter.update(message);
		}

        final String doneMessage = "Done downloading.";
        this.logger.info(doneMessage);
        this.reporter.update(doneMessage);        

    }



}
